### PR TITLE
Add commands for toggling DevCenterMode and environment listing

### PR DESCRIPTION
**Pull request by bwateratmsft**
Fixes https://github.com/Azure/azure-dev/issues/2762

## Environment listing
image
image

## Enable DevCenterMode
image
image

## Disable DevCenterMode
image
image

### CODE REVIEW

File: ext/vscode/src/commands/devCenterMode.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export function enableDevCenterMode(context: IActionContext): Promise<void> {
+    return setDevCenterMode(context, true);
+}
+
+export async function disableDevCenterMode(context: IActionContext): Promise<void> {
+    return setDevCenterMode(context, false);
+}
+
+async function setDevCenterMode(context: IActionContext, enable: boolean): Promise<void> {
```
Comment (ellismg) [inline ext/vscode/src/commands/devCenterMode.ts:17]: Just curious, @savannahostrowski, would we want to send a telemetry event for turning dev center on and off?

**Review by ellismg — APPROVED** [review]
LGTM - One question for @wbreza since I have this memory that we use `on` and `off` not `true` and `false`, but I could be wrong there.

All the other stuff looks good to me.

File: ext/vscode/src/commands/devCenterMode.ts (hunk 1 above)
Comment (bwateratmsft) [reply ext/vscode/src/commands/devCenterMode.ts:17]: Fortunately telemetry comes for free with commands using the `@microsoft/vscode-azext-utils` package. So already have telemetry on this. 👍

File: ext/vscode/src/commands/devCenterMode.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    const azureCli = await createAzureDevCli(context);
+    const command = azureCli.commandBuilder
+        .withArg('config')
+        .withArg('set')
+        .withArg('devCenterMode')
+        .withArg(enable ? 'true' : 'false');
+
+    await execAsync(command.build(), azureCli.spawnOptions());
+    void vscode.window.showInformationMessage(vscode.l10n.t('Azure Dev CLI\'s Dev Center mode has been {0}.', enable ? vscode.l10n.t('enabled') : vscode.l10n.t('disabled')));
```
Comment (savannahostrowski) [inline ext/vscode/src/commands/devCenterMode.ts:26]: This should be "Azure Developer CLI" (per marketing guidance).

**Review by savannahostrowski — APPROVED** [review]
One small comment but tried this out and LGTM

File: ext/vscode/src/commands/devCenterMode.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    return setDevCenterMode(context, false);
+}
+
+async function setDevCenterMode(context: IActionContext, enable: boolean): Promise<void> {
+    const azureCli = await createAzureDevCli(context);
+    const command = azureCli.commandBuilder
+        .withArg('config')
+        .withArg('set')
+        .withArg('devCenterMode')
+        .withArg(enable ? 'true' : 'false');
```
[the remark this answers is not in the document]
Comment (bwateratmsft) [reply ext/vscode/src/commands/devCenterMode.ts:23]: Spoke with @wbreza offline, it's "on" and "off".

**Pull request merged**